### PR TITLE
(fix) Updates for latest Rust/Iron/Hyper

### PR DESCRIPTION
**Pull request by AnthIste**
Most of the changes are just around signatures and namespacing

### CODE REVIEW

File: src/cache_handler.rs [hunk 1]
```diff
@@ -45,20 +59,19 @@ impl StaticWithCache {
...
         if requested_path.should_redirect(request) {
-            return self.static_handler.call(request);
+            return self.static_handler.handle(request);
         }
 
         match requested_path.get_file() {
             Some(file) => {
                 let last_modified_time = match file.stat() {
-                    Err(error) => return Err(box FileError(error) as IronError),
+                    Err(error) => return Err((IronError::new(FileError(error), status::Ok))),
```
Comment (reem) [inline src/cache_handler.rs]: Should also be `status::NotFound`, since no file was found.

File: src/lib.rs [hunk 2]
```diff
@@ -1,18 +1,20 @@
 #![crate_name = "static"]
 #![deny(missing_docs)]
 #![deny(warnings)]
-#![feature(phase)]
+
+#![feature(core)]
```
Comment (reem) [inline src/lib.rs]: Collapse to one line

File: src/lib.rs [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
+#![feature(collections)]
+#![feature(io)]
+#![feature(path)]
 
 //! Static file-serving handler.
 
-#[phase(plugin)]
-extern crate regex_macros;
-extern crate regex;
+// extern crate regex_macros;
```
Comment (reem) [inline src/lib.rs]: Just delete

File: src/static_handler.rs [hunk 4]
```diff
@@ -49,22 +49,20 @@ impl Handler for Static {
...
-            Some(path) =>
-                Ok(Response::new()
-                       .set(Status(status::Ok))
-                       // Won't panic because we know the file exists from get_file
-                       .set(Body(path))),
+            Some(path) => Ok(Response::with((status::Ok, path))),
             None =>
                 // If no file is found, return a 404 response.
-                Ok(Response::new().set(Status(status::NotFound)).set(Body("File not found")))
+                Ok(Response::with((status::NotFound, "File not found")))
```
Comment (reem) [inline src/static_handler.rs]: "File not found" should be appended to all `status::NotFound`s or none at all.

File: src/cache_handler.rs [hunk 5]
```diff
@@ -29,14 +44,13 @@ impl StaticWithCache {
     // Defers to the static handler, but adds cache headers to the response.
     fn defer_and_cache(&self, request: &mut Request,
                        modified: Timespec) -> IronResult<Response> {
-        use hyper::header::{CacheControl, LastModified};
-        use hyper::header::common::cache_control::CacheDirective::{Public, MaxAge};
+        use hyper::header::{CacheControl, CacheDirective, LastModified};
```
Comment (reem) [inline src/cache_handler.rs]: `hyper::header` is available from iron as `iron::headers`.

File: Cargo.toml [hunk 6]
```diff
@@ -8,6 +8,7 @@ repository = "https://github.com/iron/static"
 license = "MIT"
 
 [dependencies]
+hyper="*"
```
Comment (reem) [inline Cargo.toml]: No need to depend on hyper, headers are available in `iron::headers`

**Comment by AnthIste** [comment]
I've gone and made the changes you suggested. Is it OK to no longer box the errors that are being returned?

**Comment by AnthIste** [comment]
Any updates on this? Or is this effectively superseded by #24 ?

**Comment by reem** [comment]
It is superseded, sorry I was not very attentive here.

**Pull request closed without merging**